### DISCUSSION TITLE
Question: Large Data

### DISCUSSION THREAD

**Discussion by JackTan25** [milvus-io/milvus#28400] [Q&A and General discussion]
For milvus, the main index and fastest index is HNSW, but it's just a in-memory index. And I have 20 billion vectors (dimention is 1024), so How much memory should I need at least for this? Or is there other advice?

**Comment by xiaofan-luan** [comment]
use dskann index, or directly use zilliz cloud capacity optimized instance

**Comment by yhmo** [comment]
Use this tool to estimate required resources: https://milvus.io/tools/sizing
